parcels/application_kernels/advection.py

```diff
@@ -14,6 +16,15 @@
 ]
 
 
+def UxAdvectionEuler(particle, fieldset: UXFieldSet, time):
+    """Advection of particles using Explicit Euler (aka Euler Forward) integration.
+    on an unstructured grid."""
+    vel, ei = fieldset.eval(["u","v"],time,particle.depth,particle.lat,particle.lon, particle.ei[0])
```

Comment:
Push particle.ei to eval and select which igrid.

Also, push the xarray/uxarray "wrapper" down to the `field` class